lib/cog_api/fake/roles.ex

```diff
@@ -75,4 +75,11 @@ defmodule CogApi.Fake.Roles do
   defp permissions_match?(%Permission{} = permission, %Permission{} = match) do
     permission.namespace == match.namespace && permission.name == match.name
   end
+
+  defp prepare_role(role) do
```

Comment:
This is the wrong side of the association. We don't store the `group_ids` here, this has to look those up instead.

This is more like the `has_one` side of the equation if that helps.